### PR TITLE
[BEAM-11103] Add blog post for go 2.40 release

### PR DESCRIPTION
**Pull request by damccorm**
This blog post highlights the work we've been doing in the 2.40 beam release since its a particularly significant release for us

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
LGTM. No notes. I look forward to us publishing this!

**Comment by sachinag** [comment]
LGTM, amazing work @damccorm

**Comment by TheNeuralBit** [comment]
Should we merge this with @pabloem's Beam 2.40.0 release announcement?

**Comment by damccorm** [comment]
> Should we merge this with @pabloem's Beam 2.40.0 release announcement?

I don't think so - this is specifically highlighting that 2.40 is a big release for the Go SDK and goes beyond the scope of the typical release notes. Most of these features do appear in the release notes (with less descriptiveness), but IMO its nice to have a coherent package that we can advertise a little more easily and succinctly.

**Comment by sachinag** [comment]
I agree with Danny that a separate post makes sense.  Honestly, a
separate post on runInference is warranted too :D

**Comment by TheNeuralBit** [comment]
Fair, it does look like we have lots of precedent for separate blog posts about specific launches that come out in parallel with release announcements (including one of my own! https://beam.apache.org/blog/dataframe-api-preview-available/)

File: website/www/site/content/en/blog/go-2.40.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+gains, check out the [registration doc page](https://pkg.go.dev/github.com/apache/beam/sdks/go/pkg/beam/register).
+
+# What's Next?
+
+Moving forward, we remain focused on improving the streaming experience and
+leveraging generics to improve the Sdk. Specific improvements we are considering
+include adding [State & Timers](https://beam.apache.org/documentation/programming-guide/#state-and-timers)
+support, introducing a Go expansion service so that Go DoFns can be used in other
+languages, wrapping more Java and Python IOs so that they can be easily used in Go,
+and exploring the use of generics to simplify the user experience. As always,
```
Comment (robertwb) [inline website/www/site/content/en/blog/go-2.40.md:67]: "exploring the use of generics to simplify the user experience" is somewhat redundant with "leveraging generics to improve the Sdk" above.

**Pull request merged**